src/core/include/openvino/pass/pattern/op/label.hpp

```diff
@@ -53,13 +53,13 @@ class OPENVINO_API Label : public Pattern {
               },
               OutputVector()) {}
 
-    Label(const element::Type& type, const PartialShape& s, ValuePredicate pred)
+    Label(const element::Type& type, const PartialShape& s, const ValuePredicate& pred)
         : Label(type, s, pred, OutputVector{}) {}
```

Comment:
```suggestion
    Label(const element::Type& type, const PartialShape& s, ValuePredicate pred)
        : Label(type, s, std::move(pred), OutputVector{}) {}
```
Check similar changes where API functions changed to avoid change signatures and move can work better when functor created in-place.